### ISSUE TITLE
Improve info.repository_yaml is not None error message

### ISSUE THREAD

**Issue by Aylr** [dagster-io/dagster#207]
## STR

1. have this in your `repository.yml` file:
```
repository:
  module: backbone
  fn: define_repository
```
2. Have a `local.yml` file that looks like this:
```
context:
  name: production
  config:
    redshift:
      user: "xxx"
      password: "xxx"
      host: "localhost"
      port: 5432
      database: "xxx"
```
2. Run the following and see the stack trace
```
$ dagster pipeline execute create_common_and_derived_tables_pipeline --env local.yml
Traceback (most recent call last):
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/bin/dagster", line 10, in <module>
    cli(obj={})
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/lib/python3.6/site-packages/dagster/cli/pipeline.py", line 276, in execute_command
    pipeline = load_pipeline_from_target_info(pipeline_target_info)
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/lib/python3.6/site-packages/dagster/cli/dynamic_loader.py", line 271, in load_pipeline_from_target_info
    mode_data = create_pipeline_loading_mode_data(info)
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/lib/python3.6/site-packages/dagster/cli/dynamic_loader.py", line 192, in create_pipeline_loading_mode_data
    check.invariant(info.repository_yaml is not None)
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/lib/python3.6/site-packages/dagster/check/__init__.py", line 124, in invariant
    raise_with_traceback(CheckError('Invariant failed.'))
  File "/Users/taylor/repos/cricket-health-201806/analysis_backbone/.venv/lib/python3.6/site-packages/future/utils/__init__.py", line 419, in raise_with_traceback
    raise exc.with_traceback(traceback)
dagster.check.CheckError: Invariant failed.
```

**Comment by bhcastleton** [comment]
thanks for putting this in @Aylr, I had family show up last night and never got back to it.

**Comment by schrockn** [comment]
fixed in 0.2.3
